core/src/main/java/cz/o2/proxima/scheme/SchemaDescriptors.java

```diff
@@ -103,7 +155,24 @@ private SchemaDescriptors() {
    * @return Primitive type descriptor
    */
   public static PrimitiveTypeDescriptor<Long> longs() {
-    return primitives(AttributeValueType.LONG);
+    return longs(
+        object -> {
+          if (object instanceof Long) {
```

Comment:
+1, can we do the same for `integers()`?

Reply:
Maybe instead of `instanceof` might be metter `Long.class.isAssignableFrom(object.getClass())`